python/setup.py

```diff
@@ -26,7 +25,7 @@
 
 import versioneer
 
-install_requires = ["numba", "cython"]
+install_requires = ["numba", "cython", "cuda-python"]
```

Comment:
I think we should take this opportunity to add version minimums on these requirements.

Reply:
Would  the same  constraints  as in the  `environments/*.yml` files work?

Reply:
Kinda sucks that we need to keep so many things in sync...

Reply:
Actually I wonder if we should hold off on this and address it separately. There are other issues like why is Cython in `install_requires`? They seems worthwhile to solve, but a bit out-of-scope for the changes here

Reply:
> Would the same constraints as in the `environments/*.yml` files work?

Yes. It would be good to double check by testing an environment with those minimal pinnings. (numba==0.49.0, Cython==0.29, cuda-python==11.5).

Given how many significant bugfixes were introduced in the 0.29 line of Cython, I would not be surprised if the minimum pinning of Cython is actually supposed to be 0.29.x for some x > 0. (Note that Cython doesn’t actually have a 0.29.0 release. It’s just tagged as 0.29.)

Reply:
@jakirkham @shwina Yes, I’d be fine with punting on the versioning questions here. I didn’t expect there to be so many changes required.

Reply:
Raised as issue ( https://github.com/rapidsai/rmm/issues/945 ). Please feel free to add anything else there